### DISCUSSION TITLE
Github

### DISCUSSION THREAD

**Discussion by soundd2** [community/community#102833] [Actions] [answered]
### Select Topic Area

Question

### Body

I have created workflow as below, but when I tried to go to actions and tried to run , I am not sure why it is showing as fail and eventually it not showing run option. 

Please suggest fix. Thank you.

![image](https://github.com/community/community/assets/94822959/84583d7f-c876-40a6-8974-fe4465912ab9)

![image](https://github.com/community/community/assets/94822959/2ef06af0-0c0e-4f72-8f62-8ee4602b55f5)

**Comment by phuccoder2000** [comment]


**Reply by soundd2** [reply]
Hi @phuccoder2000 

Thank you for the response.


I am using following YAML, it appears to be correct for me, I am not certain about syntax problem.

Could you please correct me where is wrong in the below yaml

`# Build pipeline

name: webstore-build 

on: workflow_dispatch:

jobs:
 build:
     name: 'build and publish our app'
     runs-on: ubuntu-latest 
     steps: 
          - name: checkout 
            uses: actions/checkuout@v3.5.3 
          - name: 'Restore'
            run: dotnet restore
          
          - run: dotnet build --configuration Release
            name: build 
          
          - run: dotnet publish --configuration Release -o ./webstore 
            name: publish

          - name: Upload artifact
            uses: actions/upload-artifact@v3.1.2
            with:
             name: webstore
             path: ./webstore

           `

**Reply by soundd2** [reply]
Hi @phuccoder2000 

I have updated the code as below as per your suggestion.

![image](https://github.com/community/community/assets/94822959/fee32f35-6180-4b52-a08e-d2d0b49526e4)


but seems to be error with trigger, 

![image](https://github.com/community/community/assets/94822959/6dbebde5-c687-42db-81bf-33ae53f0f031)

Could you please suggest further

**Comment by rwader-swi** [comment, marked as answer]
@soundd2 
<img width="557" alt="image" src="https://github.com/community/community/assets/138435075/269eb024-0534-4c35-bd7c-bfbeaf393305">
indentations aren't correct. Also try putting workflow_dispatch on next line with 2 spaces.

**Reply by soundd2** [reply]
@phuccoder2000  rwader,


![image](https://github.com/community/community/assets/94822959/9731b8bb-6266-44fe-9594-158e764502a1)


I adjusted syntax but still not working.
I have verfied in YAML lint page also 


![image](https://github.com/community/community/assets/94822959/f900fdb3-81c6-4d78-b0ae-efcdd2c84477)

**Reply by rwader-swi** [reply]
have you kept any non-yml files in .github/workflows? try removing them.  According to https://stackoverflow.com/questions/63014238/github-action-no-event-triggers-defined-in-on.
Also make sure this workflow is both on default branch (mostly its master) and your current working branch

**Reply by soundd2** [reply]
Hi Rwader.

You can see where it is located and code structure. I am using highlighted file 

![image](https://github.com/community/community/assets/94822959/d2382668-4172-40b3-b754-5782b815705f)


I am still seeing same error.

**Reply by rwader-swi** [reply]
you are supposed to put it inside the workflows folder 🙂. If your code is in the one outside the workflows folder, then move the code to inside workflows and delete the outer one.
And use the below code only 
<img width="506" alt="image" src="https://github.com/community/community/assets/138435075/38ccfb33-c6fb-4af3-acac-cddbf8ddbfbc">

**Reply by phuccoder2000** [reply]
@soundd2 
It seems that the error "No event triggers defined in `on` " is likely caused by the absence of an event trigger in the `on` section. In your provided YAML, you have `workflow_dispatch` as the trigger event, and that's correct

![image](https://github.com/community/community/assets/62839458/6e9ffbab-75a8-4ff6-99d4-33c760f80166)

**Reply by soundd2** [reply]
Hi Rwader,

Thank you, it seems it worked. but it gave different error this time, 
it is my account only, so what I should to own that artifact and to get rid of below error 

![image](https://github.com/community/community/assets/94822959/83ccaacb-135c-487d-8aa9-9757b3d4493c)

**Reply by phuccoder2000** [reply]
workflow_dispatch:
      repository:
         owner: shoundd2

**Reply by soundd2** [reply]
Hi @phuccoder2000 

Repository is not recognizing 

![image](https://github.com/community/community/assets/94822959/c4bf3494-ee29-4565-a29c-7f267a47631e)

**Reply by soundd2** [reply]
COuld you please paste the code

**Reply by rwader-swi** [reply]
@soundd2  Okay great. You'll need a token for that checkout. Can you try printing ${{github.token}}? If its blank then you'll need to generate one. 
@phuccoder2000 he only is the owner of his repository i guess, i don't think that'll create a problem.

Print the token the way @phuccoder2000 checked for {{github.repository_owner}}. **Don't print token if its not your personal project but some enterprise thing 🙂.**
Also if there's error add `shell: bash` just above `run: |`

**Reply by soundd2** [reply]
Hi Rwader,

Can you give me screenshot what exactly to be added/amended in the below YAML, I am new to this workflows stuff.

![image](https://github.com/community/community/assets/94822959/c0d85a00-5b6d-48a0-9bbc-c5f1979515c7)

**Reply by rwader-swi** [reply]
if there's error, try removing `shell: bash` line, or try putting quotes around {{github.token}}
<img width="276" alt="image" src="https://github.com/community/community/assets/138435075/1330c749-b8c6-4ac8-849b-b081e8724968">

**Reply by rwader-swi** [reply]
Also the push and workflow_dispatch are still not indented well

**Reply by soundd2** [reply]
HI Rwader,

I still see same error related to access error even I remove shell: bash

![image](https://github.com/community/community/assets/94822959/008098fa-de36-4ca0-91fd-32b636462d22)

**Reply by rwader-swi** [reply]
This wasn't supposed to fix your error. We're just checking if the token exists. 

I just forked your repo and made some changes. [https://github.com/rwader-swi/webstore](https://github.com/rwader-swi/webstore)
Sorry I had no option, its better than writing here 🙂 
Its working now [https://github.com/rwader-swi/webstore/actions/runs/7725253134/job/21059067456](https://github.com/rwader-swi/webstore/actions/runs/7725253134/job/21059067456) 🎉 
I cancelled the run before build and publish stages, so don't worry.
You can copy paste that code (don't pull or it will show me as author). Then I'll delete the fork.

**Reply by rwader-swi** [reply]
changes I did: 
deleted the wrong syntax main.yml file, and the one outside the workflow folder 🙂
token was there, its redacted. You forgot to put $ :)

**Reply by soundd2** [reply]
HI Rwader.

Thank you.

I copied the code, where is the rerun option?

![image](https://github.com/community/community/assets/94822959/00ba13a8-2533-4c7d-a79f-e74028a668d7)

2 questions:

1. always .yml file should be inside workflows /github/workflows folder? am I correct?
2. workflow dispatch indicated manual but where is the run button to run manaul?
3. if I put  on: push , so for this trigger workflow runs when commit happens from any branch , is this correct?
